src/components/Button/Button.tsx

```diff
@@ -1,16 +1,28 @@
-/** Данный тип описывает пропсы компонента */
-export type ButtonProps = {
-  // "?" означает, что пропс необязательный
-  // e: React.MouseEvent – объект события
-  onClick?: (e: React.MouseEvent) => void;
-  children: React.ReactNode; // реакт-элемент
+import Loader from 'components/Loader';
+import cn from 'classnames';
+import Text from 'components/Text';
+import styles from './button.module.scss';
+
+export type ButtonProps = React.ButtonHTMLAttributes<HTMLButtonElement> & {
+  /** Состояние загрузки */
+  loading?: boolean;
+  /** Текст кнопки */
+  children: React.ReactNode;
 };
 
-// указываем тип пропсов комопнента в дженерике React.FC<>
-const Button: React.FC<ButtonProps> = ({ onClick, children }) => (
-  <button className="button" onClick={onClick}>
-    {children}
-  </button>
-);
+const Button: React.FC<ButtonProps> = ({ className, loading, disabled, children, ...props }) => {
+  return (
+    <button
+      className={cn(className, styles.button, disabled && styles.disabled)}
```

Comment:
Передаваемый пропс `className` нужно вставлять в последнюю очередь, так как внешние стили должны перебивать внутренние, если будет конфликт стилей
```suggestion
      className={cn(styles.button, disabled && styles.disabled, className)}
```

Reply:
+